DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/CourseDelegates/Index.cshtml

```diff
@@ -48,39 +39,52 @@
     </div>
 
     @if (Model.Courses.Any() && Model.CourseDetails != null) {
+      @if (!Model.CourseDetails.Active) {
+        <partial name="~/Views/TrackingSystem/Shared/_InactiveCourseInset.cshtml" />
+      }
+
+      <div class="display-none" id="selected-customisation-Id">
+        @Model.CustomisationId
+      </div>
+
       <div class="nhsuk-grid-row">
         <form method="get" role="search">
           <div class="nhsuk-grid-column-full nhsuk-form-group">
             <label class="nhsuk-label--s nhsuk-u-margin-bottom-1">Choose course</label>
-            <select class="nhsuk-select nhsuk-search__input nhsuk-search__input--withdropdown course-dropdown"
-                    asp-for="CustomisationId"
-                    asp-items="Model.Courses"
-                    aria-label="Choose course">
-            </select>
-            <button class="nhsuk-search__submit course-submit" type="submit">
-              Select
-            </button>
+            <div class="course-dropdown-container">
+              <select class="nhsuk-select nhsuk-search__input nhsuk-search__input--withdropdown course-dropdown"
+                      asp-for="CustomisationId"
+                      asp-items="Model.Courses"
+                      aria-label="Choose course">
+              </select>
+              <button class="nhsuk-search__submit course-submit" type="submit">
+                Select
+              </button>
+            </div>
           </div>
         </form>
       </div>
 
-      @if (!Model.CourseDetails.Active) {
-        <partial name="~/Views/TrackingSystem/Shared/_InactiveCourseInset.cshtml" />
-      }
+      <partial name="SearchablePage/Configurations/_ThreeQuarterWidthSortAndFilter" model="Model.CourseDetails" />
 
       <div class="nhsuk-grid-row">
         <div class="nhsuk-grid-column-full">
-          @if (!Model.CourseDetails.Delegates.Any()) {
+          @if (Model.CourseDetails.NoDataFound) {

```

Comment:
If left how it was, edge cases would arise where the JS does not correctly display items when loading the page with filters in the cookie. If I recall, certain sections the JS interacts with would not be rendered. 

I think the different results from different orders are because of the JS pagination overriding the page load pagination. If there are no delegates whatsoever on the course then No delegates found is displayed. However, if the user then throws a new filter on, the JS code cannot find any results and we don't seem to pass any information to it to distinguish _no base items_ to _no filtered results_ so the generic number of results crops up. Hopefully, if someone loads the page and there are no delegates before any filters are applied, they will understand what that means.

I've also changed the wording to make it slightly more obvious.

Reply:
Ah, so we can't actually know if there any delegates enrolled on the course if the initial request we send for the page has a filter in it - they might be filtered out server-side. I think that's sufficient grounds to leave this as is. We should mention it to HEE. Do we expect to demo this again?

Reply:
I don't believe we plan on demoing it again. It was demoed a while ago around the time it first went into review and there haven't been any noticeable changes since.